armada/handlers/armada.py

```diff
@@ -218,14 +219,23 @@ def wait_until_deployed(self):
                 LOG.info("{}-{}".format(name, status))
                 # wait for pod status to be True
                 # check every second until timeout?
-                for x in range(0, self.timeout):
+                isDeployed = Event()
+                Timer(self.timeout, (_ for _ in ()).throw(RuntimeError("Build timeout: {}".format(name)))).start()
```

Comment:
I know this line is hacky but it's far and away the least convoluted way to throw an exception after the timeout. this (http://eventlet.net/doc/modules/timeout.html) seems to accomplish the same thing in a prettier way but would also add a dependency. all I can find based on std libs (aside from this) would require a new class just for the timeout. I'm open to other solutions @gardlt